dataverse-webapp/src/test/java/edu/harvard/iq/dataverse/externaltools/ExternalToolServiceBeanTest.java

```diff
@@ -1,224 +1,262 @@
 import edu.harvard.iq.dataverse.persistence.user.ApiToken;
-import org.junit.Test;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.CsvSource;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
 
 import javax.json.Json;
 import javax.json.JsonObjectBuilder;
 import java.util.ArrayList;
+import java.util.Calendar;
+import java.util.Date;
 import java.util.List;
 
-import static org.junit.Assert.assertEquals;
-import static org.junit.Assert.assertNotNull;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
 
+@ExtendWith(MockitoExtension.class)
 public class ExternalToolServiceBeanTest {
 
-    public ExternalToolServiceBeanTest() {
-    }
+    @InjectMocks
+    ExternalToolServiceBean externalToolServiceBean;
 
-    @Test
-    public void testfindAll() {
+    @Mock
+    private DataFileServiceBean datafileService;
+    @Mock
+    private AuthenticationServiceBean authService;
+    @Mock
+    private ExternalToolHandler externalToolHandler;
+    @Mock
+    private DataverseSession session;
+    @Mock
+    private EjbDataverseEngine commandEngine;
+    @Mock
+    private DataverseRequestServiceBean dvRequestService;
+    @Mock
+    private WorldMapPermissionHelper worldMapPermissionHelper;
+    @Mock
+    private CitationFactory citationFactory;
+
+    // -------------------- TESTS --------------------
+
+    @ParameterizedTest
+    @DisplayName("Should show explore tools for ingested files only if they are public")
+    @CsvSource({"false, false, false, 0",
+                "true, true, false, 0",
+                "true, false, true, 0",
+                "true, false, false, 1"})
+    void findExternalToolsByFileAndVersion(boolean released, boolean embargoed, boolean restricted, int expectedSize) {
+        // given
         DataFile dataFile = new DataFile();
         dataFile.setId(42l);
-        FileMetadata fmd = new FileMetadata();
-        DatasetVersion dv = new DatasetVersion();
-        Dataset ds = new Dataset();
-        dv.setDataset(ds);
-        fmd.setDatasetVersion(dv);
-        List<FileMetadata> fmdl = new ArrayList<FileMetadata>();
-        fmdl.add(fmd);
-        dataFile.setFileMetadatas(fmdl);
+        DatasetVersion datasetVersion = new DatasetVersion();
+        datasetVersion.setVersionState(released ? DatasetVersion.VersionState.RELEASED : DatasetVersion.VersionState.DRAFT);
+        Dataset dataset = new Dataset();
+        datasetVersion.setDataset(dataset);
+
+        Calendar calendar = Calendar.getInstance();
+        calendar.setTime(new Date());
+        calendar.add(Calendar.DAY_OF_MONTH, 1);
+        dataset.setEmbargoDate(embargoed ? calendar.getTime() : null);
```

Comment:
Wouldn't it be more descriptive if we at least called calendar variable more like `futureEmbargoExpirationDate` or something?